ex35.py

```diff
@@ -6,9 +6,11 @@ def gold_room():
 
     choice = raw_input("> ")
     try:
-        if re.match('[0-9]', choice) and int(choice) < 50:

```

Comment:
If choice can be cast to an int, then it will be an int, and you don't need to check that the string form of it looks like an int; the cast is all you need, and is also the thing that would throw a ValueError exception.